protocol/kafka_confluent/v2/message_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+ Copyright 2023 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package kafka_confluent
+
+import (
+	"context"
+	"testing"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/stretchr/testify/require"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/binding/format"
+	"github.com/cloudevents/sdk-go/v2/test"
+)
+
+var (
+	ctx            = context.Background()
+	testEvent      = test.FullEvent()
+	testTopic      = "test-topic"
+	topicPartition = kafka.TopicPartition{
+		Topic:     &testTopic,
+		Partition: int32(0),
+		Offset:    kafka.Offset(10),
+	}
+	structuredConsumerMessage = &kafka.Message{
+		TopicPartition: topicPartition,
+		Value: func() []byte {
+			b, _ := format.JSON.Marshal(&testEvent)
+			return b
+		}(),
+		Headers: []kafka.Header{{
+			Key:   "content-type",
+			Value: []byte(cloudevents.ApplicationCloudEventsJSON),
+		}},
+	}
+	binaryConsumerMessage = &kafka.Message{
+		TopicPartition: topicPartition,
+		Value:          []byte("hello world!"),
+		Headers: mustToKafkaHeaders(map[string]string{
+			"ce_type":            testEvent.Type(),
+			"ce_source":          testEvent.Source(),
+			"ce_id":              testEvent.ID(),
+			"ce_time":            test.Timestamp.String(),
+			"ce-specversion":     "1.0",
+			"ce_dataschema":      test.Schema.String(),
+			"ce_datacontenttype": "text/json",
+			"ce_subject":         "receiverTopic",
+			"ce_exta":            "someext",
```

Comment:
question: for custom extensions, generally we don't use `ce` prefix. Is this supposed to test custom extension?